app/models/user.rb

```diff
@@ -45,19 +45,36 @@ def potential_swap_users(number = 5)
   def create_potential_swaps(number = 5)
     max_attempts = number * 2
     while potential_swaps.reload.count < number
+      try_to_create_marginal_swap
+      max_attempts -= 1
       try_to_create_potential_swap
       max_attempts -= 1
       break if max_attempts <= 0
     end
   end
 
   def try_to_create_potential_swap
-    swaps = User.where(
+    swaps = complementary_voters.where("constituency_ons_id like '_%'")
+    return one_swap_from_possible_users(swaps)
+  end
+
+  def try_to_create_marginal_swap
+    swaps = complementary_voters.where(
+      { constituency_ons_id: marginal_polls.map(&:constituency_ons_id) }
+    )
+    return one_swap_from_possible_users(swaps)
+  end
+
+  private def complementary_voters
+    User.where(
       preferred_party_id: willing_party_id,
       willing_party_id: preferred_party_id
-    ).where("constituency_ons_id like '_%'")
-    offset = rand(swaps.count)
-    target_user = swaps.offset(offset).limit(1).first
+    )
+  end
+
+  private def one_swap_from_possible_users(user_query)
+    offset = rand(user_query.count)
+    target_user = user_query.offset(offset).limit(1).first
```

Comment:
```suggestion
    target_user = user_query.offset(offset).take
```

although I haven't tested that's correct.

Reply:
take turns out to be less db load. 👍 change now pushed to branch